### PR TITLE
Upgrade to .net core 2.2

### PR DESCRIPTION
**Pull request by norrismiv**
Migrate to .net core 2.2. Server runs as expected.

ISSUES:
~~Log4Net does not function properly. Alternative or workaround will be needed.~~
~~MoonSharp does not register assembly properly~~
~~Console sizing and buffering break cross-platform~~
Control Service is no longer supported and will need a rework.
MOTD returns "Generic MOTD." in the interim, pending refactor of internal APIs.

### CODE REVIEW

File: hybrasyl/Client.cs [hunk 1]
```diff
@@ -629,7 +629,7 @@ public void Redirect(Redirect redirect, bool isLogoff = false)
             var x03 = new ServerPacket(0x03);
             x03.Write(addressBytes);
             x03.WriteUInt16((ushort)redirect.Destination.Port);
-            x03.WriteByte((byte)(redirect.EncryptionKey.Length + Encoding.GetEncoding(949).GetBytes(redirect.Name).Length + 7));
+            x03.WriteByte((byte)(redirect.EncryptionKey.Length + Encoding.ASCII.GetBytes(redirect.Name).Length + 7));
```
Comment (baughj) [inline hybrasyl/Client.cs]: Why remove this? This still works on 2.2?

File: hybrasyl/Game.cs [hunk 2]
```diff
@@ -264,18 +263,18 @@ public static void Main(string[] args)
...
             // Set default logging level
-            ((log4net.Repository.Hierarchy.Hierarchy) LogManager.GetRepository()).Root.Level = Level.Info;
-                ((log4net.Repository.Hierarchy.Hierarchy) LogManager.GetRepository()).RaiseConfigurationChanged(
+            ((log4net.Repository.Hierarchy.Hierarchy) LogManager.GetRepository(Assembly.GetEntryAssembly())).Root.Level = Level.Info;
+                ((log4net.Repository.Hierarchy.Hierarchy) LogManager.GetRepository(Assembly.GetEntryAssembly())).RaiseConfigurationChanged(
                 EventArgs.Empty);
             
             // Set console buffer, so we can scroll back a bunch
-            Console.BufferHeight = Int16.MaxValue - 1;
+            // Console.BufferHeight = Int16.MaxValue - 1; //Removed for cross-platform compatibility.
```
Comment (baughj) [inline hybrasyl/Game.cs]: `BufferHeight` is supported by .NET Core

File: hybrasyl/Game.cs [hunk 3]
```diff
@@ -344,7 +343,9 @@ public static void Main(string[] args)
                         stipulationWriter.Write($"Welcome to Hybrasyl!\n\nThis is Hybrasyl (version {Assemblyinfo.Version}).\n\nFor more information please visit http://www.hybrasyl.com");
                         if (string.IsNullOrEmpty(Motd))
                         {
-                            Motd = ControlService.GetMotd();
+                            //TODO: Rework for .net core
+                            //Motd = ControlService.GetMotd();
+                            Motd = "Generic MOTD.";
```
Comment (baughj) [inline hybrasyl/Game.cs]: TBH this is fine / not a big deal since you can still set via server.msg

File: hybrasyl/Hybrasyl.csproj [hunk 4]
```diff
@@ -1,223 +1,35 @@
...
-    <DebugType>pdbonly</DebugType>
-    <Optimize>true</Optimize>
-    <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-  </PropertyGroup>
-  <PropertyGroup>
-    <ApplicationIcon>hybrasyl.ico</ApplicationIcon>
+    <TargetFramework>netcoreapp2.2</TargetFramework>
```
Comment (baughj) [inline hybrasyl/Hybrasyl.csproj]: hooray

File: hybrasyl/Client.cs (hunk 1 above)
Comment (norrismiv) [reply hybrasyl/Client.cs]: Code page 949 not found without extension which cause massive cpu usage issues.

File: hybrasyl/Game.cs (hunk 2 above)
Comment (norrismiv) [reply hybrasyl/Game.cs]: Really? It threw an error saying it wasn't supported when I made these changes.

File: hybrasyl/Game.cs (hunk 3 above)
Comment (norrismiv) [reply hybrasyl/Game.cs]: Correct.

File: hybrasyl/Hybrasyl.csproj (hunk 4 above)
Comment (norrismiv) [reply hybrasyl/Hybrasyl.csproj]: yues

File: hybrasyl/Client.cs (hunk 1 above)
Comment (baughj) [reply hybrasyl/Client.cs]: ah, yeah, I see that now `Unhandled Exception: System.NotSupportedException: No data is available for encoding 949. For information on defining a custom encoding, see the documentation for the Encoding.RegisterProvider method.` - are we sure this doesn't have any impact on the usda client?

**Comment by baughj** [comment]
🎉

File: hybrasyl/Client.cs (hunk 1 above)
Comment (norrismiv) [reply hybrasyl/Client.cs]: Yeah, I was not able to find any issues. And even after referencing some of Kyle's old data he utilized the ascii codepage

Comment (baughj) [reply hybrasyl/Client.cs]: 👍

**Comment by baughj** [comment]
🔥 💣 💯

**Comment by norrismiv** [comment]
:ear:

**Pull request merged**